src/gt4py/cartesian/config.py

```diff
@@ -18,9 +18,7 @@
 GT4PY_INSTALLATION_PATH: str = os.path.dirname(os.path.abspath(__file__))
 
 # Default paths (taken from user's environment vars when possible)
-BOOST_ROOT: str = os.environ.get(
-    "BOOST_ROOT", os.environ.get("BOOST_HOME", os.path.abspath("/usr/local"))
-)
+BOOST_ROOT: str = "DEPRECATED: NOT USED ANYMORE"
```

Comment:
What about using a `__dir__` module function like the motivation example in PEP 562: https://peps.python.org/pep-0562/#rationale ?